nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/test/java/org/apache/nifi/distributed/cache/server/map/TestDistributedMapServerAndClient.java

```diff
@@ -312,6 +354,26 @@ private DistributedMapCacheClientService createClient(final int port) throws Ini
         return client;
     }
 
+    private Thread startServerSocket(ServerSocketChannel serverSocketChannel) {
+        final Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                while (true) {
+                    try {
+                        serverSocketChannel.accept();
+                    } catch (final IOException e) {
+                        return;
+                    }
+                }
+            }
+        };
+        final Thread thread = new Thread(runnable);
+        thread.setDaemon(true);
+        thread.setName("Test Server Socket");
+        thread.start();
+        return thread;
+    }
```

Comment:
As mentioned above, it would be better to replace this approach with a test socket server using Netty, which provides a more robust approach to managing threads.

Reply:
Thabk you so much @exceptionfactory for looking at this and for your feedback.  @jlshoem graciously granted me permissions to his repository and I made a commit in response your points.

I having difficulty with the ci/cd workflow being kicked off, though.

Something about needing approval from a maintainer  https://github.com/apache/nifi/actions/runs/3206783125
Is this something you might be able to assist with?

Reply:
Thank you for the fast turnaround @exceptionfactory!